apps/api/src/schema/programs.ts

```diff
@@ -1,6 +1,28 @@
 import { z } from "@hono/zod-openapi";
 
 const programIdBase = z.string({ required_error: "programId is required" });
+const majorIdBase = z.string();
+
+export const majorQuerySchema = z.object({
+  majorId: majorIdBase.optional().openapi({
+    description: "A major ID to query majors for",
```

Comment:
This is a bit clunky; the meaning you're trying to convey is "if you specify an ID, you'll only get that major back". Saying "major" twice is a bit awkward.